src/Laravel/ApiPlatformProvider.php

```diff
@@ -478,7 +489,17 @@ public function register(): void
                 $app->make(ResourceClassResolverInterface::class)
             );
         });
+        $this->app->singleton(JsonApiSchemaFactory::class, function (Application $app) {
+            return new JsonApiSchemaFactory(
+                $app->make(SchemaFactory::class),
+                $app->make(PropertyMetadataFactoryInterface::class),
+                $app->make(ResourceClassResolverInterface::class),
+                $app->make(ResourceMetadataCollectionFactoryInterface::class),
+                $app->make(DefinitionNameFactoryInterface::class),
+            );
+        });
 
+        $this->app->bind(SchemaFactoryInterface::class, JsonApiSchemaFactory::class);
```

Comment:
at some point we may want to not load these when `jsonapi` format is not enabled, for now it doesn't impact the runtime much I propose to leave that to future improvement.